### DISCUSSION TITLE
Debug information in GraphQL errors

### DISCUSSION THREAD

**Discussion by martinvisser** [Netflix/dgs-framework#1599] [Q&A]
We've been working with `debugInfo` in our set-up and want to add some extra information to an error, which is perfectly simple with the following piece of code:

```kotlin
TypedGraphQLError.newBuilder()
    .message("Something went wrong")
    .origin("the internet")
    .errorDetail(SERVICE_ERROR)
    .debugInfo(
        mapOf(
            "code" to "HTTP_418",
            "message" to "I'm a teapot",
            "detais" to "Some details",
            "service" to "CoffeeService",
        ),
    )
    .build()
```

In other words, you can just add trivial information to `debugInfo`, because it's a `Map<String, Object>`. However, in the `GraphQLError` from the `graphql-dgs-client` this is not possible to retrieve. Here the `GraphQLError` is a Kotlin data class with `extensions`, those `extensions` contain `debugInfo`, but that's of type `GraphQLErrorDebugInfo`:

```kotlin
@JsonIgnoreProperties(ignoreUnknown = true)
data class GraphQLErrorDebugInfo(
    @JsonProperty val subquery: String = "",
    @JsonProperty val variables: Map<String, Any> = emptyMap()
)
```

So here it's no longer a `Map<String, Object>` but the object above and therefore not usable with the same flexibility.

A simple example, generated with the `TypedGraphQLError`:

```json
{
  "errors": [
    {
      "message": "Something went wrong",
      "locations": [],
      "extensions": {
        "errorType": "UNAVAILABLE",
        "origin": "the internet",
        "errorDetail": "SERVICE_ERROR",
        "debugInfo": {
          "code: "HTTP_418",
          "message": "I'm a teapot",
          "details": "Some details",
          "service": "CoffeeService",
        }
      }
    }
  ],
  "data": {..}
}
```

Results in the following if used with the DGS client:
`GraphQLError(message=Something went wrong, path=[], locations=[], extensions=GraphQLErrorExtensions(errorType=UNAVAILABLE, errorDetail=SERVICE_ERROR, origin=the internet, debugInfo=GraphQLErrorDebugInfo(subquery=, variables={}), classification=))`
![image](https://github.com/Netflix/dgs-framework/assets/3170236/36197f67-06a3-45bf-a0e0-de8e9250fb74)

Technically we could use the `variables` part of the `GraphQLErrorDebugInfo` but that feels incorrect. Is there another way we can expose this same information when using the DGS client? If not, perhaps this implementation can be updated to accommodate this?

**Comment by srinivasankavitha** [comment]
As far as I recall, the `variables` portion of this extension in GraphQLDebugInfo is intended to capture extra information in the form of a free form map. I do agree that the naming is confusing. We should be able to add a new field to capture `additionalInformation` perhaps. 

Created an issue to track this: https://github.com/Netflix/dgs-framework/issues/1600

We do accept contributions, so feel free to open a PR as well if this is urgent for you since we won't be able to prioritize this issue anytime soon.
